### DISCUSSION TITLE
想通过nginx来转发请求，但是MQ的好像不支持带后缀的nginx路由

### DISCUSSION THREAD

**Discussion by Tensor-momo** [apache/rocketmq#8308] [Q&A] [answered]
![image](https://github.com/apache/rocketmq/assets/13475926/e1af7fb5-7270-4c43-ab7a-04613502fdf0)
连接mq的代码，192.168.0.173:19999

![image](https://github.com/apache/rocketmq/assets/13475926/a3615353-b1c0-4b14-90b7-dfcc691fe1a0)
nginx的配置，
一直连不上mq所在的192.168.0.173:9876

**Comment by Tensor-momo** [comment]
如果连接mq的代码中改成   
producer.setNamesrvAddr("192.168.0.173:9876");
那么将正常连接发送消息

**Reply by YongXiaoo** [reply, marked as answer]
可以把nginx改成在网络层代理，stream模块代理TCP流量，应该就可以了

**Reply by Tensor-momo** [reply]
是的后面用tcp代理就没问题了
#stream配置
stream {
    server {
       listen 8080;
       proxy_connect_timeout 1s;
       proxy_timeout 3s;
       proxy_pass xxx.xxx.xxx.xxx:9876;
    }
}
